kyo-os-lib/shared/src/main/scala/kyo/os.scala

```diff
@@ -0,0 +1,213 @@
+        override def stderr(newStderr: ProcessOutput): Command    = self.copy(stderr = newStderr)
+        override def redirectErrorStream(value: Boolean): Command = self.copy(redirectErrorStream = value)
+
+        override def flatten: List[Command.Simple]              = List(self)
+        override def map(f: Command.Simple => Command): Command = f(self)
+
+    end Simple
+
+    case class Piped(commands: List[Command.Simple]) extends Command:
+        self =>
+        def spawnAll: List[Process] < (IOs) =
+            if commands.isEmpty then IOs(List.empty)
+            else
+                commands.tail.foldLeft(commands.head.spawn.map(p => (p :: Nil, p.stdout))) { case (acc, nextCommand) =>
+                    for
+                        a <- acc
+                        (processes, lastStdout) = a
+                        nextProcess <- nextCommand.stdin(ProcessInput.Stream(lastStdout)).spawn
+                    yield (processes ++ List(nextProcess), nextProcess.stdout)
+                }.map(_._1)
+        override def spawn: Process < (IOs) =
+            spawnAll.map(_.last)
+
+        override def cwd(newCwd: Path): Command                   = self.map(_.copy(cwd = Some(newCwd)))
+        override def env(newEnv: Map[String, String]): Command    = self.map(_.copy(env = newEnv))
+        override def stdin(newStdin: ProcessInput): Command       = self.map(_.copy(stdin = newStdin))
+        override def stdout(newStdout: ProcessOutput): Command    = self.map(_.copy(stdout = newStdout))
+        def stderr(newStderr: ProcessOutput): Command             = self.map(_.copy(stderr = newStderr))
+        override def redirectErrorStream(value: Boolean): Command = self.map(_.copy(redirectErrorStream = value))
+        override def flatten: List[Command.Simple]                = commands
+        override def map(f: Command.Simple => Command): Command   = Command.Piped(commands.map(f).map(_.flatten).flatten)
+    end Piped
+end Command
+
+case class Process(private val process: JProcess) extends java.io.Closeable:
+    def stdin: OutputStream                                   = process.getOutputStream
+    def stdout: InputStream                                   = process.getInputStream
+    def stderr: InputStream                                   = process.getErrorStream
+    def waitFor: Int < IOs                                    = IOs(process.waitFor())
+    def waitFor(timeout: Long, unit: TimeUnit): Boolean < IOs = IOs(process.waitFor(timeout, unit))
+    def exitValue: Int < IOs                                  = IOs(process.exitValue())
+    def destroy: Unit < IOs                                   = IOs(process.destroy())
+    def destroyForcibly: JProcess < IOs                       = IOs(process.destroyForcibly())
+    def isAlive: Boolean < IOs                                = IOs(process.isAlive())
+    def close                                                 = IOs.run(destroy).pure
```

Comment:
Can we model the use of `Process` in terms of `Resources` instead of exposing a `close` method?